discord/invite.py

```diff
@@ -305,8 +309,11 @@ def __init__(self, *, state, data):
         self.approximate_member_count = data.get('approximate_member_count')
 
         inviter_data = data.get('inviter')
-        self.inviter = None if inviter_data is None else self._state.store_user(inviter_data)
+        self.inviter = inviter_data or self._state.store_user(inviter_data)
         self.channel = data.get('channel')
+        target_user_data = data.get('target_user')
+        self.target_user = target_user_data or self._state.store_user(target_user_data)
+        self._target_user_type = data.get('target_user_type', 0)
```

Comment:
```suggestion
        self.target_user_type = try_enum(InviteUserTarget, data.get('target_user_type', 0))
```

Constructing this all the time is relatively expensive.